internal/daemon/hostdaemon.go

```diff
@@ -222,27 +217,48 @@ func (d *HostDaemon) ListenAndServe() error {
 		} else {
 			done <- nil
 		}
+		d.log.Info("Stopping CNI server")
 		wg.Done()
 	}()
 
-	d.setupReconcilers()
 	wg.Add(1)
+	go func() {
+		d.log.Info("Starting Device Plugin server")
+		if err := d.dp.ListenAndServe(); err != nil {
+			done <- err
+		} else {
+			done <- nil
+		}
+		d.log.Info("Stopping Device Plugin server")
+		wg.Done()
+	}()
 
+	d.setupReconcilers()
 	ctx, cancelManager := context.WithCancel(ctrl.SetupSignalHandler())
+	wg.Add(1)
 	go func() {
 		d.log.Info("Starting manager")
-
 		if err := d.manager.Start(ctx); err != nil {
 			done <- err
 		} else {
 			done <- nil
 		}
+		d.log.Info("Stopping manager")
 		wg.Done()
 	}()
 
-	err = <-done
+	// Run functions that handle registering of servers with services (e.g. kubelet)
+	err = d.dp.RegisterWithKubelet()
```

Comment:
Why don't we do this inside the synchronous part of the dp?

Reply:
We need to make sure we have started serving before registering with kubelet.

Reply:
Agree, but this seems like something that is a detail that should be done inside the dp.

Reply:
Should we have some kind of Post Serve callback infrastructure?

Reply:
Dicussed offline. Waiting for final changes